backend/py_neuro/routers/utils.py

```
@@ -0,0 +1,330 @@
+                        img_file.write(image_bytes)
+
+                    image_info = models.ExtractedImageInfo(filename=image_filename, page_number=page_num + 1)
+                    all_images_info.append(image_info) # Add to all list first
+
+                    # Pre-filter the saved image
+                    if pre_filter_image(image_save_path):
+                         filtered_images_info.append(image_info) # Add to filtered list if it passes
+                    else:
+                         # Optionally delete the filtered-out image file immediately
+                         try: os.remove(image_save_path)
+                         except OSError: pass
```

Ruff (0.8.2)

Use `contextlib.suppress(OSError)` instead of `try`-`except`-`pass`

Replace with `contextlib.suppress(OSError)`

(SIM105)